exonum/src/api/error.rs

```diff
@@ -16,10 +16,118 @@
 
 use failure::Fail;
 use serde::Serialize;
+
+use std::collections::HashMap;
 use std::io;
 
 use crate::node::SendError;
 
+/// List of possible HTTP response codes.
+#[derive(Debug, Serialize)]
+pub enum HttpCode {
+    #[allow(missing_docs)]
+    BadRequest = 400,
+    #[allow(missing_docs)]
+    NotFound = 404,
+    #[allow(missing_docs)]
+    NotImplemented = 501,
+}
+
+/// API HTTP error struct.
+#[derive(Fail, Debug, Serialize)]
+pub struct ApiError {
+    /// HTTP error code.
+    #[serde(skip)]
+    pub http_code: HttpCode,
+    /// A URI reference to the documentation or possible solutions for the problem.
+    #[serde(rename = "type")]
+    pub error_type: String,
```

Comment:
Since the type already contains `*Error` in its name, I think that we can name this field just `type` without any ambiguity.

Reply:
`type` is a Rust keyword 🙃

Reply:
I think something like `docs_uri` could be better; IMO, `type` (or `error_type`) is not very descriptive in this context.

Reply:
Ah, always these limitations get me :upside_down_face: